core/include/traccc/clusterization/impl/measurement_creation.ipp

```diff
@@ -83,14 +96,12 @@ TRACCC_HOST_DEVICE inline void fill_measurement(
         m.surface_link = mod.surface_link;
         // normalize the cell position
         m.local = mean;
-        // normalize the variance
-        m.variance[0] = var[0] / totalWeight;
-        m.variance[1] = var[1] / totalWeight;
```

Comment:
I can't comment on that line unfortunately, but as long as we're removing the division by `totalWeight`, let's remove the whole `if (totalWeight > 0.)` check (from line 77) as well. Since that thing was just a menace so far. :frowning: Basically a cluster with a total weight of `0.` should've been checked for with an assertion, not an `if` statement. Since such clusters would've caused havoc in other ways as well. :thinking: You also removed the equivalent check from the GPU code anyway. :wink:

(We might still want to add `assert(totalWeight > 0.)`, as that should really be an assertion at this point. Since cells below some limit will always be discarded.)